:book:added supported labels and annotations refrence document (#7418)
kubernetes-sigs/cluster-api: docs/book/src/reference/labels_and_annotations.md

| unsafe.topology.cluster.x-k8s.io/disable-update-class-name-check   | It can be used to disable the webhook check on update that disallows a pre-existing Cluster to be populated with Topology information and Class.  |
| cluster.x-k8s.io/cluster-name   | It is set on nodes identifying the name of the cluster the node belongs to.  |
|cluster.x-k8s.io/cluster-namespace    | It is set on nodes identifying the namespace of the cluster the node belongs to.   |
| cluster.x-k8s.io/machine   | It is set on nodes identifying the machine the node belongs to.   |
|  cluster.x-k8s.io/owner-kind  |  It is set on nodes identifying the owner kind.   |
| cluster.x-k8s.io/owner-name   | It is set on nodes identifying the owner name.   |

> | cluster.x-k8s.io/paused   | It can be applied to any Cluster API object to prevent a controller from processing a resource. Controllers working with Cluster API objects must check the existence of this annotationon the reconciled object. |

Reviewer: nit:

```suggestion
| cluster.x-k8s.io/paused   | It can be applied to any Cluster API object to prevent a controller from processing a resource. Controllers working with Cluster API objects must check the existence of this annotation on the reconciled object. |
```

|  cluster.x-k8s.io/cloned-from-name  | It is the infrastructure machine annotation that stores the name of the infrastructure template resource that was cloned for the machine. This annotation is set only during cloning a template. Older/adopted machines will not have this annotation.   |
| cluster.x-k8s.io/cloned-from-groupkind   | It is the infrastructure machine annotation that stores the group-kind of the infrastructure template resource that was cloned for the machine. This annotation is set only during cloning a template. Older/adopted machines will not have this annotation.   |
|  cluster.x-k8s.io/skip-remediation  | It is used to mark the machines that should not be considered for remediation by MachineHealthCheck reconciler.   |
|  cluster.x-k8s.io/managed-by  | It can be applied to InfraCluster resources to signify that some external system is managing the cluster infrastructure. Provider InfraCluster controllers will ignore resources with this annotation. An external controller must fulfill the contract of the InfraCluster resource. External infrastructure providers should ensure that the annotation, once set, cannot be removed.  |

> |  topology.cluster.x-k8s.io/dry-run  | It is an annotation that gets set on objects by the topology controller only during a server side dry run apply operation. It is used for validating update webhooks for objects which get updated by template rotation (e.g. InfrastructureMachineTemplate).   When the annotation is set and the admission request is a dry run, the webhook should deny validation due to immutability. By that the request will succeed (without any changes to the actual object because it is a dry run) and the topology controller  will receive the resulting object. |

Reviewer: ```suggestion
|  topology.cluster.x-k8s.io/dry-run  | It is an annotation that gets set on objects by the topology controller only during a server side dry run apply operation. It is used for validating update webhooks for objects which get updated by template rotation (e.g. InfrastructureMachineTemplate). When the annotation is set and the admission request is a dry run, the webhook should deny validation due to immutability. By that the request will succeed (without any changes to the actual object because it is a dry run) and the topology controller will receive the resulting object. |
```

> |  machine.cluster.x-k8s.io/exclude-node-draining  | It explicitly skips node draining if set.  |

Reviewer: looks like volume detach skipping [annotation](https://github.com/kubernetes-sigs/cluster-api/pull/6413/files#diff-3a52e8a9da96eb854daeab0cfdf3e442fa0c34e142af2c3720a98a441a9cf3c9R36-R38) is missing right after this one in this list:

```
|  machine.cluster.x-k8s.io/exclude-wait-for-node-volume-detach  | It explicitly skips the waiting for node volume detaching if set.  |
```

Reviewer 2: Has been added now

**Supported Labels:**

> |label     |note     |

Reviewer 2: ```suggestion
| Label     | Note     |
```
nit: same below for annoations

|:--------:|:--------:|

> | cluster.x-k8s.io/cluster-name| this label is set on machines linked to a cluster and external objects(bootstrap and infrastructure providers) |

Reviewer 3: ```suggestion
| cluster.x-k8s.io/cluster-name| this label is set on machines linked to a cluster and external objects (bootstrap and infrastructure providers) |
```

Author: Did it.

| topology.cluster.x-k8s.io/owned| this label is set on all the object which are managed as part of a ClusterTopology. |

> |topology.cluster.x-k8s.io/deployment-name | this label is set on the generated  MachineDeployment objects to track the name of the MachineDeployment topology it represents. |

Reviewer 4: ```suggestion
|topology.cluster.x-k8s.io/deployment-name | this label is set on the generated MachineDeployment objects to track the name of the MachineDeployment topology it represents. |
```

| cluster.x-k8s.io/provider| this label is set on components in the provider manifest. this label allows to easily identify all the components belonging to a provider. the clusterctl tool uses this label for implementing provider's lifecycle operations. |
| cluster.x-k8s.io/watch-filter | this label can be applied to any Cluster API object. Controllers which allow for selective reconciliation may check this label and proceedwith reconciliation of the object only if this label and a configured value is present. |

> | cluster.x-k8s.io/interruptible|this label is used to mark the nodes that run on interruptible instances. |

Reviewer 4: ```suggestion
| cluster.x-k8s.io/interruptible| this label is used to mark the nodes that run on interruptible instances. |
```

|cluster.x-k8s.io/control-plane | this label is set on machines or related objects that are part of a control plane. |

> | cluster.x-k8s.io/set-name| this label is set on machines if they're controlled by MachineSet. |

Reviewer 3: ```suggestion
| cluster.x-k8s.io/set-name| this label is set on machines if they're controlled by a MachineSet. |
```

| cluster.x-k8s.io/deployment-name| It is set on machines if they're controlled by a MachineDeployment. |
| machine-template-hash| It is applied to Machines in a MachineDeployment containing the hash of the template. |
<br>

**Supported Annotations:**

|annotation     | note     |

> |:--------:|:--------:|

Reviewer 5: Could we have the `note` column left-justified? The centred text makes it difficult to read the longer paragraphs (especially true for those with dyslexia and similar)

Reviewer 2: Yes, please!

Author: Yeah sure.

|:--------:|:--------:|

> | topology.cluster.x-k8s.io/managed-field-paths   |  this annotation is used to store the list of paths managed by the topology controller. changes to those paths will be considered authoritative.  |

Reviewer 3: ```suggestion
| topology.cluster.x-k8s.io/managed-field-paths   |  This annotation is used to store the list of paths managed by the topology controller. changes to those paths will be considered authoritative.  |
```

Could you capitalize the first letter of sentences throughout?

Author: did it.

> | topology.cluster.x-k8s.io/managed-field-paths   |  this annotation is used to store the list of paths managed by the topology controller. changes to those paths will be considered authoritative.  |

Reviewer 3: Could you note that this is deprecated? Same for other labels which are listed as deprecated.

Author: Did it.

| unsafe.topology.cluster.x-k8s.io/disable-update-class-name-check   |   this annotation can be used to disable the webhook check on update that disallows a pre-existing Cluster to be populated with Topology information and Class.  |

> | cluster.x-k8s.io/cluster-name   |  this annotation is set on nodes identifying the name of the cluster the node belongs to.  |

Author: got it. thanks @killianmuldoon for your suggestions.

> |cluster.x-k8s.io/cluster-namespace    | this annotation is set on nodes identifying the namespace of the cluster the node belongs to   |

Reviewer 4: ```suggestion
|cluster.x-k8s.io/cluster-namespace    | this annotation is set on nodes identifying the namespace of the cluster the node belongs to.   |
```

> | cluster.x-k8s.io/machine   | this annotation is set on nodes identifying the machine the node belongs to   |

Reviewer 4: ```suggestion
| cluster.x-k8s.io/machine   | this annotation is set on nodes identifying the machine the node belongs to.   |
```

> |  cluster.x-k8s.io/owner-kind  |  this annotation is set on nodes identifying the owner kind   |

Reviewer 4: ```suggestion
|  cluster.x-k8s.io/owner-kind  |  this annotation is set on nodes identifying the owner kind.   |
```

> | cluster.x-k8s.io/owner-name   | this annotation is set on nodes identifying the owner name    |

Reviewer 4: ```suggestion
| cluster.x-k8s.io/owner-name   | this annotation is set on nodes identifying the owner name.    |
```

> | topology.cluster.x-k8s.io/managed-field-paths   |  It is used to store the list of paths managed by the topology controller. Changes to those paths will be considered authoritative.  <br> NOTE: Managed field depends on the last reconciliation of a managed object; this list can change during the lifecycle of an object, depending on how the corresponding template + patch/variable changes over time. <br> NOTE: The topology controller is only concerned about managed paths in the spec, given that we are dropping spec. From the result to reduce verbosity of the generated annotation.<br> NOTE: Managed paths are relevant only for unstructured objects where it is not possible to easily discover which fields have been set by templates + patches/variables at a given reconcile. Instead, it is not necessary to store managed paths for typed objets (e.g. Cluster, MachineDeployments) given that the topology controller explicitly sets a well-known, immutable list of fields at every reconcile.<br> Deprecated: Topology controller is now using server side apply and this annotation will be removed in a future release.  |

Reviewer 2: @fabriziopandini Should we drop this annotation from the docs?

As of v1.2 we are not using it anymore and it's hard to understand.

I would argue documenting it is more confusing to users than that it helps them in some way.

| cluster.x-k8s.io/paused   |  this annotation can be applied to any Cluster API object to prevent a controller from processing a resource. Controllers working with Cluster API objects must check the existence of this annotationon the reconciled object. |

> |   cluster.x-k8s.io/disable-machine-create | this annotation can be used to signal a MachineSet to stop creating new machines. It is utilized in the OnDelete MachineDeploymentStrategy to allow the MachineDeployment controller to scale down  older MachineSets when Machines are deleted and add the new replicas to the latest MachineSet.    |

Reviewer 4: ```suggestion
|   cluster.x-k8s.io/disable-machine-create | this annotation can be used to signal a MachineSet to stop creating new machines. It is utilized in the OnDelete MachineDeploymentStrategy to allow the MachineDeployment controller to scale down older MachineSets when Machines are deleted and add the new replicas to the latest MachineSet.    |
```

> |  cluster.x-k8s.io/delete-machine  | this annotation marks control plane and worker nodes that will be given priority for deletion when KCP or a machineset scales down. This annotation is given top priority on all delete policies.    |

Reviewer 4: ```suggestion
|  cluster.x-k8s.io/delete-machine  | this annotation marks control plane and worker nodes that will be given priority for deletion when KCP or a MachineSet scales down. This annotation is given top priority on all delete policies.    |
```

|  cluster.x-k8s.io/skip-remediation  |  this annotation is used to mark the machines that should not be considered for remediation by MachineHealthCheck reconciler.   |
|  cluster.x-k8s.io/managed-by  |   this annotation can be applied to InfraCluster resources to signify that some external system is managing the cluster infrastructure. Provider InfraCluster controllers will ignore resources with this annotation. An external controller must fulfill the contract of the InfraCluster resource. External infrastructure providers should ensure that the annotation, once set, cannot be removed.  |
|  topology.cluster.x-k8s.io/dry-run  |  this annotation is an annotation that gets set on objects by the topology controller only during a server side dry run apply operation. It is used for validating update webhooks for objects which get updated by template rotation (e.g. InfrastructureMachineTemplate).   |

> |  machine.cluster.x-k8s.io/exclude-node-draining  |  this annotation explicitly skips node draining if set  |

Reviewer 4: ```suggestion
|  machine.cluster.x-k8s.io/exclude-node-draining  |  this annotation explicitly skips node draining if set.  |
```
